Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/D7AbsoluteCrossSections.py

```diff
@@ -0,0 +1,404 @@
+# Mantid Repository : https://github.com/mantidproject/mantid
+#
+# Copyright &copy; 2018 ISIS Rutherford Appleton Laboratory UKRI,
+#   NScD Oak Ridge National Laboratory, European Spallation Source,
+#   Institut Laue - Langevin & CSNS, Institute of High Energy Physics, CAS
+# SPDX - License - Identifier: GPL - 3.0 +
+
+from mantid.api import AlgorithmFactory, NumericAxis, PropertyMode, \
+    PythonAlgorithm, WorkspaceGroupProperty, WorkspaceGroup
+from mantid.kernel import Direction, EnabledWhenProperty, FloatBoundedValidator, \
+    LogicOperator, PropertyCriterion, StringListValidator
+
+from mantid.simpleapi import *
+
+from scipy.constants import physical_constants
+import numpy as np
+import math
+
+
+class D7AbsoluteCrossSections(PythonAlgorithm):
+
+    @staticmethod
+    def _max_value_per_detector(ws):
+        if isinstance(mtd[ws], WorkspaceGroup):
+            max_values = np.zeros(shape=(mtd[ws][0].getNumberHistograms(),
+                                         mtd[ws].getNumberOfEntries()))
+            for entry_no, entry in enumerate(mtd[ws]):
+                max_values[:, entry_no] = entry.extractY().T
+        else:
+            max_values = mtd[ws].extractY().T
+        return np.amax(max_values, axis=1)
+
```

Comment:
You should also add seeAlso in all the 3 algorithms to refer to each other, so basically the full list should be the YIG, the reduction and the x-sections.